napari/_qt/layer_controls/qt_labels_controls.py

```diff
@@ -261,38 +261,23 @@ def __init__(self, layer):
         color_layout.addWidget(self.colorBox)
         color_layout.addWidget(self.selectionSpinBox)
 
-        # grid_layout created in QtLayerControls
-        # addWidget(widget, row, column, [row_span, column_span])
-        self.grid_layout.addLayout(button_row, 0, 0, 1, 4)
-        self.grid_layout.addWidget(QLabel(trans._('label:')), 1, 0, 1, 1)
-        self.grid_layout.addLayout(color_layout, 1, 1, 1, 3)
-        self.grid_layout.addWidget(QLabel(trans._('opacity:')), 2, 0, 1, 1)
-        self.grid_layout.addWidget(self.opacitySlider, 2, 1, 1, 3)
-        self.grid_layout.addWidget(QLabel(trans._('brush size:')), 3, 0, 1, 1)
-        self.grid_layout.addWidget(self.brushSizeSlider, 3, 1, 1, 3)
-        self.grid_layout.addWidget(QLabel(trans._('blending:')), 5, 0, 1, 1)
-        self.grid_layout.addWidget(self.blendComboBox, 5, 1, 1, 3)
-        self.grid_layout.addWidget(self.renderLabel, 6, 0, 1, 1)
-        self.grid_layout.addWidget(self.renderComboBox, 6, 1, 1, 3)
-        self.grid_layout.addWidget(QLabel(trans._('color mode:')), 7, 0, 1, 1)
-        self.grid_layout.addWidget(self.colorModeComboBox, 7, 1, 1, 3)
-        self.grid_layout.addWidget(QLabel(trans._('contour:')), 8, 0, 1, 1)
-        self.grid_layout.addWidget(self.contourSpinBox, 8, 1, 1, 1)
-        self.grid_layout.addWidget(QLabel(trans._('n edit dim:')), 9, 0, 1, 1)
-        self.grid_layout.addWidget(self.ndimSpinBox, 9, 1, 1, 1)
-        self.grid_layout.addWidget(QLabel(trans._('contiguous:')), 10, 0, 1, 1)
-        self.grid_layout.addWidget(self.contigCheckBox, 10, 1, 1, 1)
-        self.grid_layout.addWidget(
-            QLabel(trans._('preserve\nlabels:')), 11, 0, 1, 2
+        self._populate_grid(
+            (button_row,),
+            (...,),
+            ('label:', color_layout),
```

Comment:
another quick question... this now looks a lot like a QFormLayout (label, widget).  Do you know whether that definitely wouldn't work here?

Reply:
Real answer: I didn't know it existed :P

Hindsight answer: Grid is a bit more versatile. The `button_row` from the code above, for example, takes 2 columns and 2 lines, which it seems wouldn't work with `QFormLayout`. We also can have things that are wider than 2 columns, or things that do not have a label (several examples of this in the code).

On the other hand, we could probably migrate to `QFormLayout` if we simplify some things, and maybe that's a good idea for the future :P

Reply:
Using a `QFormLayout` would be really nice. I think `button_row` might be the only place where this fails, and those button rows might be a bit funny anyway. If we could get everything in a `QFormLayout` and not need the elipses that would be huge.

One quick fix option might be to create a vertical layout, put the buttons at the top and then QFormLayout with everything else below it?

What do you think @tlambert03 @brisvag?

Reply:
If we're sure we won't need more complex behavour, that sounds loke a good plan to me.

Reply:
> If we're sure we won't need more complex behavour, that sounds loke a good plan to me.

i think constraints here are good